hbase-server/src/main/java/org/apache/hadoop/hbase/util/TableDescriptorChecker.java

```diff
@@ -184,19 +190,13 @@ public static void sanityCheck(final Configuration c, final TableDescriptor td)
       }
 
       // check in-memory compaction
-      try {
-        hcd.getInMemoryCompaction();
-      } catch (IllegalArgumentException e) {
-        warnOrThrowExceptionForFailure(logWarn, e.getMessage(), e);
-      }
+      warnOrThrowExceptionForFailure(logWarn, hcd::getInMemoryCompaction);
```

Comment:
I added the above `warnOrThrowExceptionForFailure` overload which accepts a ThrowingRunnable. I modified all of the below checks to use that, so just some minor extra diff around try/catch handling.